clang/lib/Driver/ToolChains/Repo.h

```diff
@@ -0,0 +1,75 @@
+//===--- Repo.h - Repo ToolChain Implementations ----------------*- C++ -*-===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef LLVM_CLANG_LIB_DRIVER_TOOLCHAINS_REPO_H
+#define LLVM_CLANG_LIB_DRIVER_TOOLCHAINS_REPO_H
+
+#include "Linux.h"
+#include "clang/Driver/Tool.h"
+#include "clang/Driver/ToolChain.h"
+
+namespace clang {
+namespace driver {
+namespace tools {
+namespace repo {
+
+class LLVM_LIBRARY_VISIBILITY Linker : public Tool {
+public:
+  Linker(const ToolChain &TC) : Tool("repo::Linker", "rld", TC) {}
+
+  bool hasIntegratedCPP() const override { return false; }
+  bool isLinkJob() const override { return true; }
+
+  void ConstructJob(Compilation &C, const JobAction &JA,
+                    const InputInfo &Output, const InputInfoList &Inputs,
+                    const llvm::opt::ArgList &TCArgs,
+                    const char *LinkingOutput) const override;
+};
+} // namespace repo
+} // namespace tools
+
+namespace toolchains {
+
+class LLVM_LIBRARY_VISIBILITY RepoMuslToolChain : public ToolChain {
```

Comment:
I think that this class needs some explanation along the lines of what you told me during our phone call. Something to explain why "repo" and "musl" are bound together even though the only "repo" change is to use rld rather than an ELF linker.

Reply:
Added in the commit [0a629b99](https://github.com/SNSystems/llvm-project-prepo/pull/158/commits/0a629b99bb27274ce786c619db54e1a59d902947).